### PR TITLE
Update blueprints to import modules directly

### PR DESCRIPTION
**Pull request by HeroicEric**
## TL;DR;

This does two things:

- Adds more tests to the adapter & model blueprints
- Updates blueprints to import modules directly rather than access them through `DS`

I originally added the extra tests to cover the existing behavior before I changed it. If you'd rather have those split out into a separate PR LMK.

### Details

- Imports `JSONAPIAdapter` module directly in blueprint
- Imports `JSONAPISerializer` module directly in blueprint
- Imports `Transform` module directly in blueprint
- Imports `attr` module directly in model blueprints when needed
- Imports `belongsTo` and `hasMany` modules directly in blueprints when needed

### CODE REVIEW

**Comment by pangratz** [comment]
FYI, I have an somewhat accompanying PR #4023 open, which would make your adapter test `adapter when is named "application"` obsolete.

The failing tests should be fixed once a new release for `ember-cli-blueprint-test-helpers` is cut, as pointed out by @trabus here https://github.com/emberjs/data/pull/4023#issuecomment-169152370.

This looks very good to me :+1: ! Nice change importing modules directly!

**Comment by HeroicEric** [comment]
@pangratz turns out, the tests were failing because of a typo `importStatment !== importStatement` :smile: 

Would you like me to remove the `adapter when is named "application"` test?

**Comment by pangratz** [comment]
No, you can leave it in I guess. This PR will likely be merged soon, so I will update the test in #4023 once this is the case.

File: blueprints/model/index.js [hunk 1]
```diff
@@ -35,26 +39,44 @@ module.exports = {
...
     var needsDeduplicated = needs.filter(function(need, i) {
       return needs.indexOf(need) === i;
     });
 
+    if (shouldImportAttr) {
+      importStatements.push('import attr from \'ember-data/attr\';');
+    }
+
+    if (shouldImportBelongsTo && shouldImportHasMany) {
+      importStatements.push('import { belongsTo, hasMany } from \'ember-data/relationships\';');
```
Comment (bmac) [inline blueprints/model/index.js]: Out of scope for this pr but I've been thinking it may be useful to add a path that allows users to import `belongsTo`, `hasMany` and `attr` all on the same line.

Comment (HeroicEric) [inline blueprints/model/index.js]: :+1:

**Comment by HeroicEric** [comment]
This should wait until https://github.com/emberjs/data/issues/4075 is resolved

**Comment by HeroicEric** [comment]
^^ was fixed by https://github.com/emberjs/data/pull/4091

**Comment by bmac** [comment]
Thanks @HeroicEric
